### ISSUE TITLE
Add option to ignore escape

### ISSUE THREAD

**Issue by NSoiffer** [arnog/mathlive#57]
## Description
The expected behavior for screen readers for edit areas is that Escape removes focus from the current editable field. mathlive maps `Escape` to `\` and there is no way to override this. 

There should be a configuration option that makes `Esc` not be handled by mathlive so that a calling application that cares about accessibility can deal with it. Alternatively, perhaps `Esc` should not be mapped at all -- it doesn't really add anything since it is the same number of keystrokes to type the `Esc` key as it is to type the `\` key. That would help with mathlive's generic accessibility.

The later fix is trivial to do. In `onKeydown` i editor-keyboard.js, add the one line below involving "Escape":
```
        if (!compositionInProgress && 
            e.code !== 'Escape' &&
            e.code !== 'ControlLeft' &&
            e.code !== 'MetaLeft') {
            keydownEvent = e;
            keypressEvent = null;
            return handlers.keystroke(keyboardEventToString(e), e);
        }
```

**Comment by arnog** [comment]
I think you can override the behavior of ESC already. 

When creating the MathField, provide a onKeystroke handler. In this handler, you can intercept and capture the ESC key, or react to it being pressed by focusing/blurring something else. 

```javascript
const mathfield = MathLive.makeMathField(document.getElementById('math-editor'), {
    onKeystroke: function(key) {
        if (key === 'Esc') {
            focusAdjacentElement(1);
            return false;
        }
        return true;
    }
});
```
with:
```javascript
function focusAdjacentElement(dir) {
    if (document.activeElement /* && document.activeElement.form */) {
        const focussableElements = 'a:not([disabled]), button:not([disabled]), textarea:not([disabled]), input[type=text]:not([disabled]), [tabindex]:not([disabled]):not([tabindex="-1"])';
        const focussable = Array.prototype.filter.call(
            document.querySelectorAll(focussableElements),
            function(element) {
                return element.offsetWidth > 0 || 
                    element.offsetHeight > 0 || 
                    element === document.activeElement
            });
        let index = focussable.indexOf(document.activeElement) + dir;
        if (index < 0 ) index = focussable.length - 1;
        focussable[index].focus();
    }
}
```
Let me know if this doesn't work for you. I'm happy to add a config option for this, but I would rather avoid doing it if what's there already works.

**Comment by NSoiffer** [comment]
Thanks. I didn't read the comments thoroughly -- I thought it ate all the
keystrokes and didn't realize the return value would allow mathlive to do
its things. Works like a charm.

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>
